File: Array/121bestTimetoBuyandSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        low = float("inf")
        res = 0
        for i in prices:
            if i < low:
                low = i
            elif i - low > res:
                res = i - low
            else:
                pass
        return res

File: Array/121BestTimetoBuyandSellStock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0, buy = INT_MAX;
        for(int i=0; i<prices.size(); i++){
            if(prices[i] < buy) buy = prices[i];
            else res = max(res, prices[i]-buy);
        }
        return res;
    }
};